analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/SymbolicExecution/SETestContext.cs

```diff
@@ -47,10 +47,13 @@ internal class SETestContext
             CreateCS(methodBody, additionalParameters, null, additionalChecks);
 
         public static SETestContext CreateCS(string methodBody, string additionalParameters, string localFunctionName, params SymbolicCheck[] additionalChecks) =>
-            new(ClassCodeCS(methodBody, additionalParameters), AnalyzerLanguage.CSharp, additionalChecks, localFunctionName);
+            new(ClassCodeCS(methodBody, additionalParameters, null), AnalyzerLanguage.CSharp, additionalChecks, localFunctionName);
+
+        public static SETestContext CreateCS(string methodBody, string additionalParameters = null, string localFunctionName = null, string additionalTypes = null, params SymbolicCheck[] additionalChecks) =>
```

Comment:
What about `CreateCSWithAddtitionalTypes`, like `CreateCSLambda` below? Make `addtitionalTypes` nonoptional and remove unneeded arguments

Reply:
Introducing CreateCSWithAddtitionalTypes has the disadvantage of loosing the functionality of additionalParameters, localFunctionName, and additionalChecks.

Reply:
When looking at it now, you were basically looking for `CreateCSMethod`. It's exactly the same (except the types are nested in `Sample` (and we can ignore that nesting). So we can remove the new one.